tmail-backend/jmap/extensions/src/main/java/com/linagora/tmail/james/jmap/contact/OpenPaasContactsConsumer.java

```diff
@@ -92,29 +96,43 @@ public Flux<AcknowledgableDelivery> delivery() {
 
     private void messageConsume(AcknowledgableDelivery ackDelivery, String messagePayload) {
         Mono.just(messagePayload)
-            .map(message -> gson.fromJson(message, OpenPaasContactMessage.class))
+            .<ContactAddedRabbitMqMessage>handle((message, sink) -> {
+                try {
+                    sink.next(objectMapper.readValue(message, ContactAddedRabbitMqMessage.class));
+                } catch (JsonProcessingException e) {
+                    sink.error(new RuntimeException(e));
+                }
+            })
             .handle((msg, sink) -> {
                 handleMessage(msg, sink);
                 ackDelivery.ack();
-            });
+            })
+            .onErrorResume(e -> {
+                LOGGER.error("Failed to consume OpenPaaS added contact message", e);
+                return Mono.empty();
+            }).subscribe();
```

Comment:
Do not subscribe but make the method return a `Mono<Void>` (your operation) rather than `void`